helm-service/controller/deployment_handler.go

```diff
@@ -190,6 +183,27 @@ func (h *DeploymentHandler) getDeploymentURIs(e keptnv2.EventData) ([]string, []
 	return nil, nil, nil
 }
 
+func getPortOfService(service *corev1.Service) (int32, bool) {
+	lowestPort := int32(math.MaxInt32)
+	foundPort := false
+	for _, port := range service.Spec.Ports {
+		if port.Protocol == corev1.ProtocolTCP && port.Port < lowestPort {
+			lowestPort = port.Port
+			foundPort = true
+		}
+	}
+	// if no port explicitly marked as TCP port could be found, take the lowest port number of all specified ports
+	if !foundPort {
+		for _, port := range service.Spec.Ports {
+			if port.Port < lowestPort {
+				lowestPort = port.Port
+				foundPort = true
+			}
+		}
+	}
```

Comment:
I guess this is because of this definition in the carts-db service?
https://github.com/keptn/examples/blob/8c8334de4ca74e5977ba3ec5d1defa1644db3f58/onboarding-carts/carts-db/templates/carts-db-service.yaml#L6-L12
(vs. this in carts https://github.com/keptn/examples/blob/8c8334de4ca74e5977ba3ec5d1defa1644db3f58/onboarding-carts/carts/templates/service.yaml#L8-L12 )

Reply:
exactly - in the carts-db, no protocol is specified for the service. Therefore I changed the logic to preferably use a port that has been declared as a TCP port, if available. If no such port is found, the restriction of checking the protocol is removed and the lowest available port is used.